neuralhydrology/datasetzoo/camelsind.py

```diff
@@ -0,0 +1,410 @@
+import os
+from functools import reduce
+from pathlib import Path
+from typing import List, Dict, Union
+
+import pandas as pd
+from tqdm import tqdm
+import xarray
+
+from neuralhydrology.datasetzoo.basedataset import BaseDataset
+from neuralhydrology.utils.config import Config
+
+class CamelsIND(BaseDataset):
+    """Data set class for the CAMELS IND dataset by [#]_.
+
+    For more efficient data loading during model training/evaluating, this dataset class expects the CAMELS-IND dataset
+    in a processed format. Specifically, this dataset class works with per-basin csv files that contain all 
+    timeseries data combined. Use the :func:`preprocess_camels_ind_dataset` function to process the original dataset 
+    layout into this format.
+
+    Parameters
+    ----------
+    cfg : Config
+        The run configuration.
+    is_train : bool 
+        Defines if the dataset is used for training or evaluating. If True (training), means/stds for each feature
+        are computed and stored to the run directory. If one-hot encoding is used, the mapping for the one-hot encoding 
+        is created and also stored to disk. If False, a `scaler` input is expected and similarly the `id_to_int` input
+        if one-hot encoding is used. 
+    period : {'train', 'validation', 'test'}
+        Defines the period for which the data will be loaded
+    basin : str, optional
+        If passed, the data for only this basin will be loaded. Otherwise the basin(s) are read from the appropriate
+        basin file, corresponding to the `period`.
+    additional_features : List[Dict[str, pd.DataFrame]], optional
+        List of dictionaries, mapping from a basin id to a pandas DataFrame. This DataFrame will be added to the data
+        loaded from the dataset, and all columns are available as 'dynamic_inputs', 'evolving_attributes' and
+        'target_variables'
+    id_to_int : Dict[str, int], optional
+        If the config argument 'use_basin_id_encoding' is True in the config and period is either 'validation' or 
+        'test', this input is required. It is a dictionary, mapping from basin id to an integer (the one-hot encoding).
+    scaler : Dict[str, Union[pd.Series, xarray.DataArray]], optional
+        If period is either 'validation' or 'test', this input is required. It contains the centering and scaling
+        for each feature and is stored to the run directory during training (train_data/train_data_scaler.yml).
+
+    References
+    ----------
+    .. [#] Mangukiya, N. K., Kumar, K. B., Dey, P., Sharma, S., Bejagam, V., Mujumdar, P. P., & Sharma, A. (2025). 
+    CAMELS-IND: hydrometeorological time series and catchment attributes for 228 catchments in Peninsular India. 
+    Earth System Science Data, 17(2), 461-491.
```

Comment:
I think these two lines need to be intended to the same level as e.g. the explanation of the input Parameters.
```suggestion
        CAMELS-IND: hydrometeorological time series and catchment attributes for 228 catchments in Peninsular India. 
        Earth System Science Data, 17(2), 461-491.
```